File: go/hamming/hamming.go
package hamming

import "fmt"

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, fmt.Errorf("error")
	}

	counter := 0

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			counter += 1
		}
	}

	return counter, nil
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        if (leftStrand.length != rightStrand.length) {
            throw IllegalArgumentException("left and right strands must be of equal length")
        }

        var counter = 0

        for ((i, char) in leftStrand.withIndex()) {
            if (char != rightStrand[i]) {
                counter += 1                
            }
        }

        return counter
    }
}
